pkg/handler/metricsrecorder/handler.go

```diff
@@ -0,0 +1,338 @@
+
+	gqlRequest := GraphQLRequest{Query: query}
+	jsonData, err := json.Marshal(gqlRequest)
+	if err != nil {
+		report_err := h.reportStatus(req, false, fmt.Sprintf("failed to marshall data into gql request %s", err))
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("failed to marshal request into gql request: %w", err)
+	}
+
+	host := fmt.Sprintf("%s.observe-eng.com", accountNumber)
+	url := fmt.Sprintf("https://%s/v1/meta", host)
+
+	request, reqErr := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
+	if reqErr != nil {
+		report_err := h.reportStatus(req, false, fmt.Sprintf("failed to create request %s", err))
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Authorization", fullToken)
+
+	client := &http.Client{}
+	resp, err := client.Do(request)
+	if err != nil {
+		report_err := h.reportStatus(req, false, fmt.Sprintf("error receiving response from graphql %s", err))
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("error receiving response from graphql: %w", err)
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		report_err := h.reportStatus(req, false, fmt.Sprintf("error reading response body %s", err))
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	var result GraphQLResponse
+	err = json.Unmarshal(bodyBytes, &result)
+	if err != nil {
+		report_err := h.reportStatus(req, false, fmt.Sprintf("error unmarshalling JSON from GQL response %s", err))
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("error unmarshalling JSON from GQL response: %w", err)
+	}
+
+	logger.V(4).Info("response from observe api", "result", result)
+
+	var metricSelection []MetricsListItem
+
+	variablesList := result.Data.Datasource.Variables
+
+	hasMetricsField := false
+	for _, variable := range variablesList {
+		if variable.Name == "Metrics" {
+			hasMetricsField = true
+			break
+		}
+	}
+
+	if !hasMetricsField {
+		report_err := h.reportStatus(req, false, "no metrics field in response")
+		if report_err != nil {
+			return nil, fmt.Errorf("failed to report status to cloudformation: %w, for error %w", report_err, err)
+		}
+		return nil, fmt.Errorf("no metrics field in response")
+	}
+
+	if len(result.Data.Datasource.Variables) > 0 {
```

Comment:
confused here, why are we checking for any variables? we just checked above for a specific one? also why are we just assuming the first one is the metrics one?